### PR TITLE
HIP-584: Add WipeLogic

### LINKED ISSUE #6299: HIP-584: Copy `WipeLogic`
### Problem

In order to support `WipePrecompile` we need to copy `WipeLogic`.

### Solution

- Copy `WipeLogic` in `com.hedera.services`.
- Integrate `WipeLogic` with the Store interface.
- Add javadoc describing what has changed and how we integrated it in the `mirror-node`’s code base
- Create integration tests in `ContractCallServicePrecompileTest` in `hedera-mirror-web3` and validate that the return gas estimates in the desired range with offset between 5% and 20% more than the actual gas used
### Alternatives

_No response_

### PR DESCRIPTION
**Pull request by dikel**
**Description**:
Copies WipeLogic from services

**Related issue(s)**:

Fixes #6299 

**Checklist**

- [ ] Documented (Code comments, README, etc.)
- [x] Tested (unit, integration, etc.)

### CODE REVIEW

File: hedera-mirror-web3/src/main/java/com/hedera/services/txn/token/WipeLogic.java [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+
+/**
+ * Copied Logic type from hedera-services.
+ * <p>
+ * Differences with the original:
+ * 1. Use abstraction for the state by introducing {@link Store} interface
+ * 2. validateSyntax method uses default value of true for areNftsEnabled property
+ * 3. validateSyntax executes the logic directly instead of calling TokenWipeAccessor.validateSyntax
+ */
+@Singleton
```
Comment (0xivanov) [inline hedera-mirror-web3/src/main/java/com/hedera/services/txn/token/WipeLogic.java:54]: Remove `@Singleton` and `@Inject`. Configure the class as bean in `EvmConfiguration`

File: hedera-mirror-web3/src/main/java/com/hedera/services/txn/token/WipeLogic.java [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+import javax.inject.Inject;
+import javax.inject.Singleton;
+
+/**
+ * Copied Logic type from hedera-services.
+ * <p>
+ * Differences with the original:
+ * 1. Use abstraction for the state by introducing {@link Store} interface
+ * 2. validateSyntax method uses default value of true for areNftsEnabled property
+ * 3. validateSyntax executes the logic directly instead of calling TokenWipeAccessor.validateSyntax
```
Comment (0xivanov) [inline hedera-mirror-web3/src/main/java/com/hedera/services/txn/token/WipeLogic.java:52]: Also `GlobalDynamicProperties` is replaced with `MirrorNodeEvmProperties`

File: hedera-mirror-web3/src/main/java/com/hedera/services/txn/token/WipeLogic.java [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+            return INVALID_TOKEN_ID;
+        }
+
+        if (!body.hasAccount()) {
+            return INVALID_ACCOUNT_ID;
+        }
+        return validateTokenOpsWith(
+                body.getSerialNumbersCount(),
+                body.getAmount(),
+                true,
```
Comment (0xivanov) [inline hedera-mirror-web3/src/main/java/com/hedera/services/txn/token/WipeLogic.java:136]: Should we have a separate property for `areNftEnabled`? Please share your opinion @IvanKavaldzhiev @steven-sheehy

Comment (IvanKavaldzhiev) [reply hedera-mirror-web3/src/main/java/com/hedera/services/txn/token/WipeLogic.java:99]: I believe they would always be enabled, so we can default to `true`

File: hedera-mirror-web3/src/main/java/com/hedera/services/txn/token/WipeLogic.java [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+        if (token.getType().equals(TokenType.FUNGIBLE_COMMON)) {
+            tokenModificationResult = token.wipe(accountRel, amount);
+        } else {
+            final var tokenWithLoadedUniqueTokens = store.loadUniqueTokens(token, serialNumbersList);
+            tokenModificationResult = tokenWithLoadedUniqueTokens.wipe(accountRel, serialNumbersList);
+        }
+        /* --- Persist the updated models --- */
+        store.updateToken(tokenModificationResult.token());
+        store.updateTokenRelationship(tokenModificationResult.tokenRelationship());
+        store.updateAccount(tokenModificationResult.tokenRelationship().getAccount());
```
Comment (IvanKavaldzhiev) [inline hedera-mirror-web3/src/main/java/com/hedera/services/txn/token/WipeLogic.java:81]: You should return `tokenModificationResult`

**Review by IvanKavaldzhiev — APPROVED** [review]
LGTM

**Review by IvanKavaldzhiev — DISMISSED** [review]
LGTM

**Comment by IvanKavaldzhiev** [comment]
You can add test cases for `StoreImpl` to increase code coverage.

**Review by steven-sheehy — COMMENTED** [review]
Looks good just need to fix merge conflict

**Review by IvanKavaldzhiev — APPROVED** [review]
LGTM

**Review by steven-sheehy — APPROVED** [review]

**Pull request merged**